File: Data Structures & Algorithms/unique-paths-ii/submission-1.py
class Solution:
    def uniquePathsWithObstacles(self, obstacleGrid: List[List[int]]) -> int:
        memo = {}

        m = len(obstacleGrid)
        n = len(obstacleGrid[0])
        def dfs(i,j):
            if i >= m or j >=n:
                return 0
            if obstacleGrid[i][j] == 1:
                return 0
            if i == m-1 and j == n-1:
                return 1
            if (i,j) in memo:
                return memo[(i,j)]
            
            count = dfs(i+1, j) + dfs(i, j+1)
            memo[(i,j)] = count
            return count
        return dfs(0,0)

File: Data Structures & Algorithms/is-anagram/submission-1.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!= t.length()){
            return false;
        }
        int[] count = new int[26];
        for(int i = 0; i < s.length(); i+=1){
            count[s.charAt(i) - 'a'] +=1;
            count[t.charAt(i)- 'a'] -=1;
        }
        for(int value:count){
            if(value != 0){
                return false;
            }
        }
        return true;
    }
}
